cpu/efm32_common/periph/i2c.c

```diff
@@ -0,0 +1,266 @@
+/*
+ * Copyright (C) 2016-2017 Bas Stottelaar <basstottelaar@gmail.com>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_efm32_common
+ * @{
+ *
+ * @file
+ * @brief       Low-level I2C driver implementation
+ *
+ * @author      Bas Stottelaar <basstottelaar@gmail.com>
+ *
+ * @}
+ */
+
+#include "cpu.h"
+#include "mutex.h"
+
+#include "periph_conf.h"
+#include "periph/i2c.h"
+#include "periph/gpio.h"
+
+/* emlib uses the same flags, undefine fist */
+#undef I2C_FLAG_WRITE
+#undef I2C_FLAG_READ
+
+#include "em_cmu.h"
+#include "em_i2c.h"
+
+/* guard file in case no I2C device is defined */
+#if I2C_NUMOF
+
+static volatile I2C_TransferReturn_TypeDef i2c_progress[I2C_NUMOF];
+
+static mutex_t i2c_lock[I2C_NUMOF] = {
+#if I2C_0_EN
+    [0] = MUTEX_INIT,
+#endif
+#if I2C_1_EN
+    [1] = MUTEX_INIT,
+#endif
```

Comment:
you could remove the static initializers and simply init the applicable mutex in the `i2c_init` funciton.

Reply:
Fixed and tested.